### ISSUE TITLE
Custom DerivedData folder breaks hot reloading

### ISSUE THREAD

**Issue by genkernel** [johnno1962/HotReloading#94]
I have Xcode's DerivedData folder configured to a `Custom` path.
After following the installation steps Hot reloading did not work in my proj. Neither it worked in a mentioned `SwiftUI` example project.

When i pressed cmd + s it spit out a warning-like message in the log: `Could not locate derived data. Is the project under your home directory?`
When I inspected the code that logged above message it was obvious that it was due to derived data folder not being resolved.

Long story short after reconfiguring XCode back to default DerivedData path Hot reloading started to work.
I thought you may be interested to add a note regarding this to installation docs.

Thanks you for the lightning and feel free to close this issue.

**Comment by genkernel** [comment]
One more thing.
Just having a reference to bazel in the monorepo-like directory structure where ios project lives - breaks hot reloading. Even though ios project itself does not use bazel.
I removed bazel folders that HotReloading complained about in logs and it began working.

**Comment by johnno1962** [comment]
Hi, yes if you can think of a way a custom derived data path could be supported let me know. In theory if you don't run the app and just use "standalone" injection it should work though. Re: Basel, I should really remove that "support" as I think it no longer works.

**Comment by genkernel** [comment]
Not sure if this still works: https://github.com/NSHipster/xcderiveddata
There are similar posts on the web all of which are quite old though.

Anyhow anybody also using `Custom` path and  seeing `Could not locate derived data. Is the project under your home directory?` message should be able to find this issue and know to switch to default path.

Maybe elaborate on the log message and add a sentence saying that `Custom` paths may fail to resolve even when it is under `Home` directory. e.g: mine was ~/tmp/xcode/deriveddata.

If you like an idea of an extended message lmk I can send a pull req.
Thanks again for the lib!

**Comment by johnno1962** [comment]
There doesn't seem to be a non-awful solution to this problem involving xcodebuild which is what the article you posted suggests. Would you settle for an environment variable where you specify the custom derived data path in the scheme of your app? That's not great but it would be a reasonably small change..

**Comment by genkernel** [comment]
I switched to using default DerivedPath settings. My use case for a custom path was to pipe xcode's output to ~/tmp so that I can easily clear all my temp files with 1 command.

I think adding an environment var may only complicate things more due to split brain and thus increase maintenance.

It may be best left "as is" since there probably aren't many devs with custom DerivedData path.

My 2 cents: I would change `Could not locate derived data. Is the project under your home directory?` log message by adding `if you are using custom DerivedData path - this is not supported - or refer to this note: <env-var-for-derived-data-url>`.
This will clearly guide a developer without needing to dive into lib's source code to figure out the reason for the printed log.

**Comment by johnno1962** [comment]
I couldn't replicate getting the message you were seeing but not supporting custom DD has always bugged me and I've pushed a new commit to the HotReloading project that allows you to specify a custom DerivedData path in the environment variable INJECTION_DERIVED_DATA if you want to try it out. If this fixes it or, if you're using the app rather than the HotReloading project I can roll a new release candidate.

https://github.com/johnno1962/HotReloading/commit/42a797969338797d24a74812e06c4a48ad8dfaa5

**Comment by genkernel** [comment]
I just checked it out by importing a package with `42a7979` commit's hash and confirming that setup with `INJECTION_DERIVED_DATA` worked on my side.
Thanks!
Im closing the issue with no further suggestions.

**Comment by johnno1962** [comment]
Thanks for checking it out. I'll release some time, likely when Xcode 16.comes out.
